File: StartHere/src/main/java/com/lambdaschool/starthere/models/Book.java
package com.lambdaschool.starthere.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "books")
public class Book extends Auditable{

    @Id
    @GeneratedValue
    private long bookid;

    private String booktitle;
    private String isbn;
    private int copy;

    @ManyToMany
    @JoinTable(name = "bookauthors",
            joinColumns = {@JoinColumn(name = "bookid")},
            inverseJoinColumns = {@JoinColumn(name = "authorid")})
    @JsonIgnoreProperties("books")
    private List<Author> authors = new ArrayList<>();

    public Book() {}

    public Book(String booktitle, String isbn, int copy) {
        this.booktitle = booktitle;
        this.isbn = isbn;
        this.copy = copy;
    }

    public long getBookid() {
        return bookid;
    }

    public void setBookid(long bookid) {
        this.bookid = bookid;
    }

    public String getBooktitle() {
        return booktitle;
    }

    public void setBooktitle(String booktitle) {
        this.booktitle = booktitle;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public int getCopy() {
        return copy;
    }

    public void setCopy(int copy) {
        this.copy = copy;
    }

    public List<Author> getAuthors() {
        return authors;
    }

    public void setAuthors(List<Author> authors) {
        this.authors = authors;
    }
}

File: sprint14/src/main/kotlin/com/lambdaschool/starthere/models/Book.kt
package com.lambdaschool.starthere.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

import javax.persistence.*
import java.util.ArrayList

@Entity
@Table(name = "books")
class Book : Auditable {

    @Id
    @GeneratedValue
    var bookid: Long = 0

    var booktitle: String? = null
    var isbn: String? = null
    var copy: Int = 0

    @ManyToMany
    @JoinTable(name = "bookauthors", joinColumns = [JoinColumn(name = "bookid")], inverseJoinColumns = [JoinColumn(name = "authorid")])
    @JsonIgnoreProperties("books")
    var authors: List<Author> = mutableListOf()

    constructor() {}

    constructor(booktitle: String?, isbn: String?, copy: Int) {
        this.booktitle = booktitle
        this.isbn = isbn
        this.copy = copy
    }
}
